pkg/server/commands/listusers/list_users_rpc_test.go

```diff
@@ -1250,6 +1249,139 @@ func TestListUsersWildcards(t *testing.T) {
 	tests.runListUsersTestCases(t)
 }
 
+func TestListUsersEdgePruning(t *testing.T) {
+	tests := ListUsersTests{
+		{
+			name: "valid_edges",
+			req: &openfgav1.ListUsersRequest{
+				Object:   &openfgav1.Object{Type: "document", Id: "1"},
+				Relation: "viewer",
+				UserFilters: []*openfgav1.ListUsersFilter{
+					{
+						Type: "user",
+					},
+				},
+			},
+			model: `model
+			schema 1.1
+		  type user
+
+		  type document
+			relations			  
+			  define viewer: [user]`,
+			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKey("document:1", "viewer", "user:maria"),
+			},
+			expectedUsers: []string{"user:maria"},
+		},
+		{
+			name: "impossible_edge_several_computed_relations_away",
```

Comment:
🤔 why does this say "impossible edge"?

Reply:
Because there is no edge between those two nodes, it is impossible.

Reply:
there is no edge connecting `user` and `document#viewer`, but if you look at the entire model, there is a set of edges connecting them. i would remove `impossible_edge` from the test name because to me it implies that there is no connectivity possible, but it's not the case here.

Reply:
Ah. This is a mistake, the edge is indeed possible. Updated the test name.